src/contexts/FormioContext.tsx

```diff
@@ -1,17 +1,84 @@
-import { createContext } from 'react';
+import { createContext, useState } from 'react';
+import { Formio, EventEmitter } from '@formio/js';
 
-import { UseFormioArgs, useFormio } from '../hooks/useFormio';
+type BaseConfigurationArgs = {
+	baseUrl: string;
+	projectUrl: string;
+};
+const useBaseConfiguration = ({
+	baseUrl,
+	projectUrl,
+}: BaseConfigurationArgs) => {
+	// establish basic Formio configuration
+	if (baseUrl) {
+		Formio.setBaseUrl(baseUrl);
+	}
+	if (projectUrl) {
+		Formio.setProjectUrl(projectUrl);
+	}
+
+	// TODO: this is due to a desync in event emitters between @formio/js and @formio/core; once fixed, we shouldn't need it
+	Formio.events = new EventEmitter({
+		wildcard: false,
+		maxListeners: 0,
+	});
+
+	return {
+		Formio,
+		baseUrl: Formio.baseUrl,
+		projectUrl: Formio.projectUrl,
+	};
+};
+
+const useAuthentication = () => {
+	const [token, setToken] = useState(Formio.getToken() || null);
+	const [isAuthenticated, setIsAuthenticated] = useState(!!token);
+	// listen for user events for authentication
+	Formio.events.on('formio.user', async (user: unknown) => {
+		if (user) {
+			setToken(Formio.getToken());
+			setIsAuthenticated(true);
+		} else if (isAuthenticated) {
+			await Formio.logout();
+			setToken(null);
+			setIsAuthenticated(false);
+		}
+	});
```

Comment:
I think this needs to be in a useEffect with a clean up function.  What happens if I call useAuthentication from multiple components?

Reply:
The only consumer of `useAuthentication` is the FormioContext, but I agree this could and should have a cleanup step

Reply:
same with the isAuthenticated check